helpers/utils.py

```diff
@@ -57,6 +57,12 @@ def get_env_defaults_conf_file() -> str:
     SPARK_ENV_DEFAULTS_FILE = os.environ.get('SNAP_SPARK_ENV_CONF')
     return SPARK_ENV_DEFAULTS_FILE
 
+def get_scala_shell_history_file() -> str:
+    """Returns location of .scala_history file for spark-shell"""
+    USER_HOME_DIR = pwd.getpwuid(os.getuid())[USER_HOME_DIR_ENT_IDX]
+    SCALA_HIST_FILE_DIR = os.environ.get('SNAP_USER_DATA', USER_HOME_DIR)
+    return f'{SCALA_HIST_FILE_DIR}/.scala_history'
```

Comment:
Should we allow the user to specify the name of this file directly via an env variable?

Reply:
Not sure if we need an extra env variable for just this property. 

env file and --properties-file can always include this property. Please check my suggestion above where we add our default value of scala_history file location to setup related dynamic defaults, which are parsed before env file and --properties-file.